src/detection_win.cpp

```diff
@@ -364,104 +395,111 @@ void ExtractDeviceInfo(HDEVINFO hDevInfo, SP_DEVINFO_DATA* pspDevInfoData, TCHAR
+	if (DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_FRIENDLYNAME, &DataT, (PBYTE)buf, MAX_PATH, &nSize)) {
 		resultItem->deviceName = buf;
 	}
-	else if ( DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_DEVICEDESC, &DataT, (PBYTE)buf, buffSize, &nSize))
+	else if ( DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_DEVICEDESC, &DataT, (PBYTE)buf, MAX_PATH, &nSize))
 	{
 		resultItem->deviceName = buf;
 	}
-	if (DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_MFG, &DataT, (PBYTE)buf, buffSize, &nSize)) {
+	if (DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_MFG, &DataT, (PBYTE)buf, MAX_PATH, &nSize)) {
 		resultItem->manufacturer = buf;
 	}
-	if (DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_HARDWAREID, &DataT, (PBYTE)buf, buffSize, &nSize)) {
-		// Use this to extract VID / PID
-		extractVidPid(buf, resultItem);
-	}
+	ExtractDetails(devicePath, resultItem);
 }
 
+void HandleDeviceConnect(PDEV_BROADCAST_DEVICEINTERFACE pDevInf) {
+	HDEVINFO hDevInfo = DllSetupDiGetClassDevs(&GUID_DEVINTERFACE_USB_DEVICE, NULL, NULL, DIGCF_PRESENT | DIGCF_DEVICEINTERFACE);
+	if (INVALID_HANDLE_VALUE == hDevInfo) {
+		return;
+	}
+
+	SP_DEVINFO_DATA* pspDevInfoData = (SP_DEVINFO_DATA*) HeapAlloc(GetProcessHeap(), 0, sizeof(SP_DEVINFO_DATA));
+	pspDevInfoData->cbSize = sizeof(SP_DEVINFO_DATA);
+	for(int i=0; DllSetupDiEnumDeviceInfo(hDevInfo, i, pspDevInfoData); i++) {
+		PSP_DEVICE_INTERFACE_DETAIL_DATA pspDeviceIntDetail = NULL;
+		if (!GetDeviceDetails(hDevInfo, pspDevInfoData, i, &pspDeviceIntDetail)) {
+			continue;
+		}
+
+		if (0 == _strnicmp(pDevInf->dbcc_name, pspDeviceIntDetail->DevicePath, strlen(pDevInf->dbcc_name))) {
+			TCHAR buf[MAX_PATH];
+			if (!DllSetupDiGetDeviceInstanceId(hDevInfo, pspDevInfoData, buf, sizeof(buf), NULL)) {
+				break;
+			}
+
+			WaitForSingleObject(deviceChangedSentEvent, INFINITE);
+
+			DeviceItem_t* device = new DeviceItem_t();
+
+			ExtractDeviceInfo(hDevInfo, pspDevInfoData, pDevInf->dbcc_name, &device->deviceParams);
+			AddItemToList(buf, device);
+
+			currentDevice = &device->deviceParams;
+			isAdded = true;
+
+			break;
+		}
+	}
+
+	if (pspDevInfoData) {
+		HeapFree(GetProcessHeap(), 0, pspDevInfoData);
+	}
+
+	if (hDevInfo) {
+		DllSetupDiDestroyDeviceInfoList(hDevInfo);
+	}
 
-void UpdateDevice(PDEV_BROADCAST_DEVICEINTERFACE pDevInf, WPARAM wParam, DeviceState_t state) {
+	SetEvent(deviceChangedRegisteredEvent);
+}
+
+void HandleDeviceDisconnect(PDEV_BROADCAST_DEVICEINTERFACE pDevInf) {
 	// dbcc_name:
 	// \\?\USB#Vid_04e8&Pid_503b#0002F9A9828E0F06#{a5dcbf10-6530-11d2-901f-00c04fb951ed}
 	// convert to
 	// USB\Vid_04e8&Pid_503b\0002F9A9828E0F06
-	CString szDevId = pDevInf->dbcc_name+4;
-	int idx = szDevId.ReverseFind(_T('#'));
-
-	szDevId.Truncate(idx);
+	CString szDevId = pDevInf->dbcc_name + 4;
+	szDevId.Truncate(szDevId.ReverseFind(_T('#')));
```

Comment:
I know this is legacy but what does `szDevId` stand for?

Reply:
This is the device id string. The sz prefix is standard windows variable naming convention for a zero terminated string.

https://msdn.microsoft.com/en-us/library/windows/desktop/aa378932(v=vs.85).aspx